stetho/src/main/java/com/facebook/stetho/inspector/database/DatabasePeerManager.java

```diff
@@ -107,17 +108,54 @@ private static String removeSuffix(String str, String[] suffixesToRemove) {
     Util.throwIfNull(handler);
     SQLiteDatabase database = openDatabase(databaseName);
     try {
-      Cursor cursor = database.rawQuery(query, null);
-      try {
-        return handler.handleResult(cursor);
-      } finally {
-        cursor.close();
+      String trimmedSql = query.trim();
+      String prefixSql = "";
+      if (trimmedSql.length() >= 3) {
+        prefixSql = trimmedSql.substring(0, 3).toUpperCase();
       }
+      if (prefixSql.equals("UPD") || prefixSql.equals("DEL")) {
+        return executeUpdateDelete(database, query, handler);
+      } else if (prefixSql.equals("INS")) {
+        return executeInsert(database, query, handler);
+      } else {
+        return executeRawQuery(database, trimmedSql, handler);
+      }
+
     } finally {
       database.close();
     }
   }
 
+  private <T> T executeUpdateDelete(
+      SQLiteDatabase database,
+      String query,
+      ExecuteResultHandler<T> handler) {
+    SQLiteStatement statement = database.compileStatement(query);
+    int count = statement.executeUpdateDelete();
+    return handler.handleResult((long) count, "Rows Modified");
+  }
+
+  private <T> T executeInsert(
+      SQLiteDatabase database,
+      String query,
+      ExecuteResultHandler<T> handler) {
+    SQLiteStatement statement = database.compileStatement(query);
+    long count = statement.executeInsert();
+    return handler.handleResult(count, "ID Of last inserted row");
+  }
+
+  private <T> T executeRawQuery(
+      SQLiteDatabase database,
+      String query,
+      ExecuteResultHandler<T> handler) {
+    Cursor cursor = database.rawQuery(query, null);
+    try {
+      return handler.handleResult(cursor);
```

Comment:
Out of curiosity, what was this actually doing before when `UPDATE` et al were passed?  What was the cursor?